mmv1/templates/terraform/examples/discoveryengine_chat_engine_basic.tf.erb

```diff
@@ -0,0 +1,28 @@
+resource "google_discovery_engine_data_store" "data_store" {
+  location                    = "global"
+  data_store_id               = "<%= ctx[:vars]['data_store_id'] %>"
+  display_name                = "tf-test-structured-datastore"
+  industry_vertical           = "GENERIC"
+  content_config              = "NO_CONTENT"
+  solution_types              = ["SOLUTION_TYPE_CHAT"]
+}
+
+resource "google_discovery_engine_chat_engine" "<%= ctx[:primary_resource_id] %>" {
+  engine_id = "<%= ctx[:primary_resource_id] %>"
+  collection_id = "default_collection"
+  location = "global"
+  display_name = "test desplay name"
+  industry_vertical = "GENERIC"
+  data_store_ids = [google_discovery_engine_data_store.data_store.data_store_id, google_discovery_engine_data_store.data_store_2.data_store_id]
```

Comment:
this will fail because data_store_2 doesn't exist.

Reply:
I see, so I need to do the following:
1. Example tf.erb file: correct the terraform resource blocks so it can generate the correct TF example.
2. For `[resource_discovery_engine_chat_engine_test.go](https://github.com/GoogleCloudPlatform/magic-modules/pull/9834/files#diff-a403e7f806086575b482ea5a9f40aee6822db21a5dd86ce561e42be3475b1b53)`, create two test funcs:
a) `TestAccDiscoveryEngineChatEngine_discoveryengineChatengineBasicExample_create` : this only use one test step config func to verify create
b) `TestAccDiscoveryEngineChatEngine_discoveryengineChatengineBasicExample_update`: this uses two test steps config funcs : 1. Create the initial state. 2. Update form initial state

Is that correct?